sdk/core/Azure.Core/src/Request.cs

```diff
@@ -13,36 +14,119 @@ namespace Azure.Core
     /// Represents an HTTP request. Use <see cref="HttpPipeline.CreateMessage()"/> or <see cref="HttpPipeline.CreateRequest"/> to create an instance.
     /// </summary>
 #pragma warning disable AZC0012 // Avoid single word type names
-    public abstract class Request : IDisposable
+    public abstract class Request : PipelineRequest
 #pragma warning restore AZC0012 // Avoid single word type names
     {
-        private RequestUriBuilder? _uri;
+        private RequestMethod _method;
+        private RequestUriBuilder? _uriBuilder;
+        private RequestContent? _content;
+
+        private string? _clientRequestId;
+
+        /// <summary>
+        /// Gets or sets the request HTTP method.
+        /// </summary>
+        public new virtual RequestMethod Method
+        {
+            get => _method;
+            set => SetMethodCore(value.Method);
+        }
 
         /// <summary>
         /// Gets or sets and instance of <see cref="RequestUriBuilder"/> used to create the Uri.
         /// </summary>
-        public virtual RequestUriBuilder Uri
+        public new virtual RequestUriBuilder Uri
```

Comment:
No call into base class. Is that intentional or it will be fixed in the future PR?

Reply:
Request is abstract.  Azure.Core's internal subtypes call into System.ClientModel types to implement the functionality.

Reply:
@AlexanderSher, I've updated Request a bit in https://github.com/Azure/azure-sdk-for-net/pull/42226.

For this property, the _uriBuilder field is the source of truth, and the UriCore property gets and sets it when it is updates.  For other properties, the XxCore property is the source of truth.  I've added comments in the file to clarify.

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/42226 has merged.